src/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -27,33 +26,57 @@ internal class CSharpFileBuilder : IAssemblySymbolWriter, IDisposable
         private readonly AdhocWorkspace _adhocWorkspace;
         private readonly SyntaxGenerator _syntaxGenerator;
 
-        public CSharpFileBuilder(ISymbolFilter symbolFilter, TextWriter textWriter, CSharpSyntaxRewriter syntaxRewriter)
+        private readonly IEnumerable<MetadataReference> _metadataReferences;
+
+        public CSharpFileBuilder(
+            ISymbolFilter symbolFilter,
+            TextWriter textWriter,
+            CSharpSyntaxRewriter syntaxRewriter,
+            IEnumerable<MetadataReference> metadataReferences)
         {
             _textWriter = textWriter;
             _symbolFilter = symbolFilter;
             _syntaxRewriter = syntaxRewriter;
 
             _adhocWorkspace = new AdhocWorkspace();
             _syntaxGenerator = SyntaxGenerator.GetGenerator(_adhocWorkspace, LanguageNames.CSharp);
+
+            _metadataReferences = metadataReferences;
         }
 
         /// <inheritdoc />
         public void WriteAssembly(IAssemblySymbol assembly) => Visit(assembly);
 
         private void Visit(IAssemblySymbol assembly)
         {
+            Project project = _adhocWorkspace.AddProject(ProjectInfo.Create(
+                ProjectId.CreateNewId(), VersionStamp.Create(), assembly.Name, assembly.Name, LanguageNames.CSharp));
+            project = project.AddMetadataReferences(_metadataReferences);
+
             var namespaceSymbols = EnumerateNamespaces(assembly).Where(_symbolFilter.Include);
+            var namespaceSyntaxNodes = new List<SyntaxNode>();
 
             foreach (var namespaceSymbol in namespaceSymbols)
             {
-                SyntaxNode? syntaxNode = Visit(namespaceSymbol);
+                var syntaxNode = Visit(namespaceSymbol);
```

Comment:
Typically we don't use `var` unless it's obvious from the right-hand-side what the type is.